### DISCUSSION TITLE
AXIS A8105-E problem / bug

### DISCUSSION THREAD

**Discussion by VikMgl** [AxisCommunications/discussions#238] [ACAP]
i have this intercom model AXIS A8105-E
and i create some events - > Recipients -> here i have option http command 
in this section user and password not working 
![axis](https://user-images.githubusercontent.com/132988958/237029523-0b000046-6660-4352-a9c6-a9b921275229.jpg)

**Comment by onie5** [comment]
Hi VikMigl,

Can you provide more info on why you say the user/password is not working? It is possible the vapix is not correct?

**Reply by AlexShuMg** [reply]
Hi @onie5, Me and @VikMgl are working together on this one.
@VikMgl already wrote an explanation and attached screenshots, Ill try to summarize it:

While configuring an HTTP recipient, it seems that it ignores the authentication credentials provided,
whether we config them or not, the listener (server) on the other side received GET request with no Authentication/Authorization headers whatsoever.

It might also be true for HTTPS, haven't tested yet, but we will because we need it :)

When trying to send a get request to our server we get 401 because indeed no way of authentication was received

**Reply by onie5** [reply]
@VikMgl may I know what device is your recipient? Model name would be preferred.

Thanks

**Reply by AlexShuMg** [reply]
@onie5 Its a proprietary Http WebApi, but why does it matter?
Its just an http server,  we tested with "PacketSender" and an Http listener script in python, the all prove the same, the Axis device does not include the credentials in the request.

you can verify with "PacketSender" or equivalent tool.
The test i made with PacketSender:
1. Set up PacketSender TCP Server on `192.168.1.100:8028`
2. Send a request from Axis Device to the address of  "Packet Sender" with out credentials
3. Send a request from Axis Device to the address of  "Packet Sender" with credentials
4. Send a request from Postman to the address of  "Packet Sender" with out credentials
5. Send a request from Postman to the address of  "Packet Sender" with credentials using Basic Auth
6. Send a request from Postman to the address of  "Packet Sender" with credentials using Digest Auth

the config in the Axis device:
![image](https://github.com/AxisCommunications/discussions/assets/129746233/82802e2a-857a-4f44-b3af-1b5729630514)
![image](https://github.com/AxisCommunications/discussions/assets/129746233/c4f6d505-273f-47e4-839b-b682919ae8d6)


The results:
**Axis no credentials:**
`GET / HTTP/1.1\r\nHost: 192.168.1.100:8028\r\nAccept: */*\r\n\r\n`

**Axis with credentials:**
`GET / HTTP/1.1\r\nHost: 192.168.1.100:8028\r\nAccept: */*\r\n\r\n`

**Postman no credentials:** you can see a lot more headers bit no Auth
`GET / HTTP/1.1\r\nUser-Agent: PostmanRuntime/7.32.2\r\nAccept: */*\r\nPostman-Token: 5f5cf6b4-61cb-4c0f-9d1b-c66840bdc664\r\nHost: 192.168.1.100:8028\r\nAccept-Encoding: gzip, deflate, br\r\nConnection: keep-alive\r\n\r\n`

**Postman with Basic Auth:** you can see the Authorization header
`GET / HTTP/1.1\r\n**Authorization: Basic dXNlcjoxMjM0**\r\nUser-Agent: PostmanRuntime/7.32.2\r\nAccept: */*\r\nPostman-Token: 855b8131-e1b7-4dad-999c-1c7c1a801c75\r\nHost: 192.168.1.100:8028\r\nAccept-Encoding: gzip, deflate, br\r\nConnection: keep-alive\r\n\r\n`

**Postman with Digest Auth:** you can see the Authorization header with digest algorithm
`GET / HTTP/1.1\r\nAuthorization: Digest username="user", realm="192.168.1.100", nonce="0", uri="/", algorithm="MD5", qop=auth-int, nc=0, cnonce="0", response="2993a6813134106504c7124fe40274da"\r\nUser-Agent: PostmanRuntime/7.32.2\r\nAccept: */*\r\nPostman-Token: 81667e64-1030-48a6-9781-0a7c89ddd5b7\r\nHost: 192.168.1.100:8028\r\nAccept-Encoding: gzip, deflate, br\r\nConnection: keep-alive\r\n\r\n`

The moral of the story, the Axis device ignores the username and password provided :(

Thanks for the help

**Reply by onie5** [reply]
Is it an issue with the webserver in the device? the gui indicates it is from an old firnware version.

**Reply by VikMgl** [reply]
@onie5  its new device 
 AXIS A8105-E Network Video Door Station
Firmware version: 1.65.7

**Reply by VikMgl** [reply]
https://www.axis.com/products/axis-a8105-e/support#close

**Comment by VikMgl** [comment]
Hi when i send http command to my system or to python script 
for example 
from axis http command to our system its required user and password ( we test it and find out that axis not using this fields of user and password ) for send data.
and all the time field to send http msg because our system using user and password.
i send to python http script with user and password and not receive  information about user field 


without user field 
![image](https://github.com/AxisCommunications/discussions/assets/132988958/5fdd4009-17ef-4ce7-a890-cdc88f01a581)
![image](https://github.com/AxisCommunications/discussions/assets/132988958/96314a74-658c-49fa-ba3c-8d9da4367f2e)

with user field 
![image](https://github.com/AxisCommunications/discussions/assets/132988958/606d5f49-c83e-47c9-9580-949212b3d2ff)
![image](https://github.com/AxisCommunications/discussions/assets/132988958/df3e0d68-d85f-4087-ad3b-8ba3d5ec93ea)



thanks

**Comment by VikMgl** [comment]
and this msg to our system with user and password 
![image](https://github.com/AxisCommunications/discussions/assets/132988958/e6c6124e-0b5b-4a8a-bd14-e32adbf190c7)
how u can see Unauthorized

**Comment by vivekatoffice** [comment]
*If your question remains unanswered for a period of time, please revisit it to see whether it can be improved by following the guidelines listed in [Axis support guidelines](https://github.com/orgs/AxisCommunications/discussions/13).*
